tempo_embeddings/embeddings/vector_database.py

```diff
@@ -164,91 +174,85 @@ def get_existing_collection(self, name: str) -> Optional[Collection]:
+                embeddings = [tensor.tolist() for tensor in embedded_tensors][0]
+            elif passages_need_embeddings and not self.model and not self.embedding_function:
+                raise RuntimeError("These passages need embeddings but no valid model or embedding function was provided to the Database Object")
+            else: 
+                embeddings = None
+            # embeddings = [tensor.tolist() for tensor in batch_emb] if batched_embeddings != [] else None
+            for k, p in enumerate(batch_pass):
+                pid = p.get_unique_id()
+                if pid not in seen_ids:
+                    docs.append(p.text)
+                    p.metadata["tokenized_text"] = " ".join(p.words())
+                    p.metadata["highlighting"] = p.highlighting.get_span(stringify=True) if p.highlighting else "-"
+                    metas.append(p.metadata)
+                    ids.append(pid)
+                    seen_ids.add(pid)
+                    if embeddings:
+                        p.embedding = embeddings[k]
+                    if p.embedding is not None:
+                        insert_embeds.append(p.embedding)
+            if len(insert_embeds) > 0:
+                collection.add(documents=docs, metadatas=metas, ids=ids, embeddings=insert_embeds)
             else:
-                start_slice = i * self.batch_size
-                embeds = embeddings_list[start_slice : start_slice + self.batch_size]
-                collection.add(
-                    documents=docs, embeddings=embeds, metadatas=metas, ids=ids
-                )
+                collection.add(documents=docs, metadatas=metas, ids=ids)
 
         new_count = collection.count()
         print(f"Added {new_count - num_records} new documents. Total = {new_count}")
 
     def compress_embeddings(
         self,
-        collection: Collection = None,
-        persist_in_db: bool = False,
+        passages: list[Passage],
         umap_verbose: bool = True,
         **umap_args,
     ) -> ArrayLike:
-        records = self.get_records(collection)
 
-        if len(records["ids"]) == 0:
+        if len(passages) == 0:
             return None
 
-        full_embeddings = records["embeddings"]
-        database_ids = records["ids"]
-        metadatas = records["metadatas"]
+        full_embeddings = [p.embedding for p in passages]
 
         umap = UMAP(verbose=umap_verbose, **umap_args)
         compressed = umap.fit_transform(full_embeddings)
 
-        if persist_in_db:
-            for i, m in enumerate(metadatas):
-                datapoint = list(compressed[i])
-                m["datapoint_x"] = float(datapoint[0])
-                m["datapoint_y"] = float(datapoint[1])
-            collection.update(database_ids, metadatas=metadatas)
-            print(
-                "Corresponding Datapoints (datapoint_x, datapoint_y) were saved in the database"
-            )
-
         return compressed
 
-    def get_records(
+    def get_passages(
```

Comment:
For consistency with my suggestions above, I think it makes sense to wrap the passages into a `Corpus` object here.

```suggestion
    def get_corpus(
```